plugins/restful/RestfulDataProviderDbQuery.php

```diff
@@ -99,7 +113,7 @@ public function __construct(array $plugin, \RestfulAuthenticationManager $auth_m
 
     $this->tableName = $options['table_name'];
     $this->idColumn = $options['id_column'];
-    $this->primary = empty($plugin['primary']) ? NULL : $this->primary = $plugin['primary'];
+    $this->primary = !empty($plugin['primary']) ? $options['primary'] : $options['id_column'];
```

Comment:
This will set `$primary` from the id column, in case the `'primary'` key is omitted from the plugin definition.  This is used in `update()` to skip updating the primary and id fields.